haystack/generator/transformers.py

```diff
@@ -291,3 +292,181 @@ def predict(self, query: str, documents: List[Document], top_k: Optional[int] =
+
+
+        See https://huggingface.co/transformers/main_classes/model.html?transformers.generation_utils.GenerationMixin#transformers.generation_utils.GenerationMixin
+        as well as https://huggingface.co/blog/how-to-generate
+
+        For a list of all text-generation models see https://huggingface.co/models?pipeline_tag=text-generation
+
+        **Example**
+
+        ```python
+        |     query = "Why is Dothraki language important?"
+        |
+        |     # Retrieve related documents from retriever
+        |     retrieved_docs = retriever.retrieve(query=query)
+        |
+        |     # Now generate answer from query and retrieved documents
+        |     generator.predict(
+        |        query=query,
+        |        documents=retrieved_docs,
+        |        top_k=1
+        |     )
+        |
+        |     # Answer
+        |
+        |     {'answers': [" The Dothraki language is a constructed fictional language. It's important because George R.R. Martin wrote it."],
+        |      'query': 'Why is Dothraki language important?'}
+        |
+        ```
+    """
+
+    def __init__(
+            self,
+            model_name_or_path: str,
+            input_converter: Optional[Callable] = None,
+            top_k: int = 1,
+            max_length: int = 200,
+            min_length: int = 2,
+            num_beams: int = 8,
+            use_gpu: bool = True,
+    ):
+        """
+
+
+        :param model_name_or_path: a HF model name for auto-regressive language model like GPT2, XLNet, XLM, Bart, T5
+        and others (e.g. `yjernite/bart_eli5`)
+        :input_converter: an optional Callable to prepare model input for the underlying language model specified in
+        model_name_or_path parameter
+        :param top_k: Number of independently generated text to return
+        :param max_length: Maximum length of generated text
+        :param min_length: Minimum length of generated text
+        :param num_beams: Number of beams for beam search. 1 means no beam search.
+        :param use_gpu: Whether to use GPU (if available)
+        """
+
+        self.model_name_or_path = model_name_or_path
+        self.max_length = max_length
+        self.min_length = min_length
+        self.num_beams = num_beams
+
+        if top_k > self.num_beams:
+            top_k = self.num_beams
+            logger.warning(f'top_k value should not be greater than num_beams, hence setting it to {num_beams}')
+
+        self.top_k = top_k
+
+        if use_gpu and torch.cuda.is_available():
+            self.device = torch.device("cuda")
+        else:
+            self.device = torch.device("cpu")
+
+        Seq2SeqGenerator._register_converters(model_name_or_path, input_converter)
+
+        self.tokenizer = AutoTokenizer.from_pretrained(model_name_or_path)
+        self.model = AutoModelForSeq2SeqLM.from_pretrained(model_name_or_path).to(self.device)
+        self.model.eval()
```

Comment:
just curious why this step is needed?

Reply:
The eval switch? Yeah, it's not, perhaps just to be explicit that the model is indeed in eval mode.